### PR TITLE
ScalaPB 1.0 and document use of local protoc for OSX ARM

### PR DESCRIPTION
**Pull request by ennru**
ScalaPB 1.0 (in RC right now) allows setting the `protoc` executable so that a locally built version can be used.

References #1238

### CODE REVIEW

**Comment by raboof** [comment]
I'm really looking forward to sbt-protoc 1.0.0, it will also fix running protoc without further workarounds on NixOS \o/

File: build.sbt [hunk 1]
```diff
@@ -226,8 +225,7 @@ lazy val pluginTesterJava = Project(id = "akka-grpc-plugin-tester-java", base =
     ReflectiveCodeGen.generatedLanguages := Seq("Java"),
     crossScalaVersions := Dependencies.Versions.CrossScalaForLib,
     scalaVersion := scala212,
-    ReflectiveCodeGen.codeGeneratorSettings ++= Seq("server_power_apis"),
-    Compile / ReflectiveCodeGen.protocOptions += "--include_std_types")
```
Comment (raboof) [inline build.sbt:230]: why did you remove this here? did it become the default behavior?

Comment (ennru) [reply build.sbt:230]: I tried to find something in the docs, but didn't succeed.
`protoc` doesn't accept the flag anymore.

**Comment by ennru** [comment]
This is similar to #1146

File: project/Dependencies.scala [hunk 2]
```diff
@@ -73,6 +73,10 @@ object Dependencies {
     val logback = "ch.qos.logback" % "logback-classic" % "1.2.3" % "runtime" // Eclipse 1.0
   }
 
+  object Protobuf {
+    val googleCommonProtos = "com.thesamet.scalapb.common-protos" %% "proto-google-common-protos-scalapb_0.10" % "1.18.1-0" % "protobuf"
```
Comment (ennru) [inline project/Dependencies.scala:77]: Is there a need/way to deduct the versions (0.10, 1.18.1-0) from the protoc plugin?

File: project/Common.scala [hunk 3]
```diff
@@ -74,5 +75,9 @@ object Common extends AutoPlugin {
     testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-oDF"),
     crossScalaVersions := Seq(scala212, scala213),
     mimaReportSignatureProblems := true,
-    scalafmtOnCompile := true)
+    scalafmtOnCompile := true,
+    // https://github.com/akka/akka-grpc/issues/1238
+    PB.protocExecutable := (if (protocbridge.SystemDetector.detectedClassifier() == "osx-aarch_64")
+                              file("/usr/local/bin/protoc")
```
Comment (ennru) [inline project/Common.scala:81]: This should be configurable. Hopefully ScalaPB can offer a good solution for it.

File: project/Dependencies.scala (hunk 2 above)
Comment (raboof) [reply project/Dependencies.scala:77]: not AFAICS

File: project/Common.scala (hunk 3 above)
Comment (raboof) [reply project/Common.scala:81]: it's configurable by overriding `PB.protocExecutable` - perhaps that is sufficient?

Comment (ennru) [reply project/Common.scala:81]: Yes, that should be enough. A note/pointer in the Akka gRPC docs would be good.

File: docs/src/main/paradox/buildtools/sbt.md [hunk 4]
```diff
@@ -106,7 +117,7 @@ If you want to use TLS-based negotiation on JDK 8 versions prior to
 [1.8.0_251](https://www.oracle.com/java/technologies/javase/8u251-relnotes.html),
 the server requires a special Java agent for ALPN.
  
-See the [Akka HTTP docs about HTTP/2](https://doc.akka.io/docs/akka-http/10.1/server-side/http2.html#application-layer-protocol-negotiation-alpn-))
+See the @extref:[Akka HTTP docs about HTTP/2](akka-http:server-side/http2.html#application-layer-protocol-negotiation-alpn-)
```
Comment (raboof) [inline docs/src/main/paradox/buildtools/sbt.md:120]: :+1:

**Review by raboof — APPROVED** [review]
